query-engine/js-connectors/js/neon-js-connector/src/neon.ts

```diff
@@ -127,7 +140,11 @@ class PrismaNeon implements Connector, Closeable {
     const { sql, args: values } = query
 
     try {
-      return await this.client.query(sql, values)
+      if (this._httpMode) {
+        return await this.client!.query(sql, values)
+      } else {
+        return await this.httpClient!(sql, values)
+      }
```

Comment:
While we may revisit how we run queries with either client (conditionally, depending on `this._httpMode`, similarly as to how we do it for PlanetScale), let's keep it as is for the time being